babel.config.js

```diff
@@ -1,8 +1,133 @@
+
+// 	return {
+// 		presets: [ '@wordpress/babel-preset-default' ],
+// 		plugins: [ 'babel-plugin-inline-json-import' ],
+// 	};
+// };
+
+// module.exports = function( api ) {
+// 	api.cache( true );
+
+// 	return {
+//         overrides: [
+//             {
+//                 exclude: './packages/**/test/*.native.[jt]s?(x)',
+//                 presets: [ '@wordpress/babel-preset-default' ],
+//                 plugins: [ 'babel-plugin-inline-json-import' ],
+//             },
+//             {
+//                 include: './packages/**/test/*.native.[jt]s?(x)',
+//                 presets: [
+//                     'module:metro-react-native-babel-preset',
+//                 ],
+//                 plugins: [
+//                     '@babel/plugin-proposal-async-generator-functions',
+//                     '@babel/plugin-transform-runtime',
+//                     [
+//                         'react-native-platform-specific-extensions',
+//                         {
+//                             extensions: [
+//                                 'css',
+//                                 'scss',
+//                                 'sass',
+//                             ],
+//                         },
+//                     ],
+//                 ],
+//                 env: {
+//                     development: {
+//                         plugins: [
+//                             '@babel/transform-react-jsx-source',
+//                         ],
+//                     },
+//                 },
+//             },
+//             { // Transforms JSX into JS function calls and use `createElement` instead of the default `React.createElement`
+//                 exclude: /node_modules\/react-native/,
+//                 plugins: [
+//                     [
+//                         '@babel/plugin-transform-react-jsx',
+//                         {
+//                             pragma: 'createElement',
+//                             pragmaFrag: 'Fragment',
+//                         },
+//                     ],da
+//                 ],
+//             },
+//             { // Auto-add `import { createElement } from '@wordpress/element';` when JSX is found
+//                 exclude: /node_modules\/react-native/,
+//                 plugins: [
+//                     [
+//                         './packages/babel-plugin-import-jsx-pragma',
+//                         {
+//                             scopeVariable: 'createElement',
+//                             scopeVariableFrag: 'Fragment',
+//                             source: '@wordpress/element',
+//                             isDefault: false,
+//                         },
+//                     ],
+//                 ],
+//             },
+//         ]
+// 	};
+// };
+
 module.exports = function( api ) {
 	api.cache( true );
-
 	return {
-		presets: [ '@wordpress/babel-preset-default' ],
-		plugins: [ 'babel-plugin-inline-json-import' ],
+		presets: [
```

Comment:
I think we should probably use two different babel config files. Is there a way to pass a specific file to the jest commands for instance?

Reply:
I'll try @koke's [idea](https://github.com/WordPress/gutenberg/pull/16404#issuecomment-507980760) to split the the babel.config.

I was looking for a way to pass the config file path via command but it seems to always default to the root folder where the command is executed. So changing where do we execute the command seems to be a valid option!